operator/v1/types_ingress.go

```diff
@@ -210,7 +210,7 @@ type IngressControllerStatus struct {
 	Domain string `json:"domain"`
 
 	// endpointPublishingStrategy is the actual strategy in use.
-	EndpointPublishingStrategy EndpointPublishingStrategy `json:"endpointPublishingStrategy"`
+	EndpointPublishingStrategy *EndpointPublishingStrategy `json:"endpointPublishingStrategy,omitempty"`
```

Comment:
Is it okay for the status field to be a pointer? Otherwise, until we publish our default on status, we'd report what looks like an invalid state:

```yaml
status:
  endpointPublishingStrategy:
    type: ""
```

Reply:
cc @smarterclayton

Reply:
yes that is ok to have a pointer in status.